ambry-store/src/test/java/com.github.ambry.store/LogSegmentTest.java

```diff
@@ -285,6 +297,76 @@ public void readTest() throws IOException {
     }
   }
 
+  /**
+   * Tests {@link LogSegment#readIntoDirectly(byte[], long, int)} for various cases.
+   * @throws IOException
+   */
+  @Test
+  public void readDirectTest() throws IOException {
+    assumeTrue(Platform.isLinux());
```

Comment:
can the test of the read() and readDirect() be combined?

Reply:
It's not straightforward, let me think about this.

Reply:
Combined.